packages/devtools_app/lib/src/performance/performance_controller.dart

```diff
@@ -315,16 +317,28 @@ class PerformanceController extends DisposableController
       return;
     }
 
-    if (!frame.isWellFormed &&
-        (_firstWellFormedFrameMicros == null ||
-            frame.timeFromFrameTiming.start.inMicroseconds >=
-                _firstWellFormedFrameMicros)) {
+    final bool frameBeforeFirstWellFormedFrame =
+        firstWellFormedFrameMicros == null ||
+            frame.timeFromFrameTiming.start.inMicroseconds <
+                firstWellFormedFrameMicros;
+    if (!frame.isWellFormed && !frameBeforeFirstWellFormedFrame) {
       // Only try to pull timeline events for frames that are after the first
       // well formed frame. Timeline events that occurred before this frame will
       // have already fallen out of the buffer.
       await processAvailableEvents();
     }
 
+    // If the frame is still not well formed after processing all available
+    // events, wait a short delay and try to process events again after the
+    // VM has been polled one more time.
+    if (!frame.isWellFormed && !frameBeforeFirstWellFormedFrame) {
+      _processing.value = true;
+      await Future.delayed(timelinePollingInterval, () async {
+        await processTraceEvents(allTraceEvents);
+        _processing.value = false;
+      });
+    }
+
     data.selectedFrame = frame;
```

Comment:
what if someone clicks another frame while awaiting this async operation? Add a 
_nextSelectedFrame member and skip if frame != _nextSelectedFrame.

Reply:
added `_currentFrameBeingSelected` member and a check after each await. PTAL.